src/bulk/common.ts

```diff
@@ -1093,21 +1094,16 @@ export abstract class BulkOperationBase {
       throw new MongoInvalidArgumentError('Operation must be an object with an operation key');
     }
     if ('insertOne' in op) {
-      const forceServerObjectId = shouldForceServerObjectId(this);
-      if (op.insertOne && op.insertOne.document == null) {
-        // NOTE: provided for legacy support, but this is a malformed operation
-        if (forceServerObjectId !== true && (op.insertOne as Document)._id == null) {
-          (op.insertOne as Document)._id = new ObjectId();
-        }
-
-        return this.addToOperationsList(BatchType.INSERT, op.insertOne);
-      }
+      const forceServerObjectId = this.shouldForceServerObjectId();
+      const document =
+        op.insertOne && op.insertOne.document == null
+          ? // NOTE: provided for legacy support, but this is a malformed operation
+            (op.insertOne as Document)
```

Comment:
This probably shouldn't have just been left as a comment, we should actually remove this, ideally, TS is already preventing this kind of code from existing for 3 major versions now. todo & ticket?

Reply:
I don't know if it's safe to remove - this is available in mongosh.  In mongosh, TS doesn't prevent this 🙂

Reply:
I agree, not here of course, but perhaps in a major we could right?

Reply:
I filed a ticket in the v7 epic